src/mode/behaviour/cstyle.js

```diff
@@ -311,7 +311,8 @@ var CstyleBehaviour = function(options) {
                 var line = session.doc.getLine(cursor.row);
                 var nextLine = session.doc.getLine(cursor.row + 1);
                 var indent = this.$getIndent(line);
-                if (/\s*\*/.test(nextLine)) {
+                const beforeCursor = line.substring(0, cursor.column);
+                if (/\s*\*/.test(nextLine) && /\*/.test(beforeCursor)) {
```

Comment:
Wouldn't this match ` a * b  /**comment*/` with cursor before the comment ?

Reply:
Hmm, I don't think so because we still check if there is an `*` in next line, so I guess it would match:
```
a * b /**<cursor>comment*/
*
```
Which would be turned into 
```
a * b /**
* <cursor>comment*/
*
```
I think, but let me double check this. Nice test example!!

Reply:
Yeah so actually nothing cool happens because state is not `"doc-start"`, so it just goes into new line.
```

Reply:
sorry, my bad for including `*/` in the end of the example. In default js example of this pull request https://raw.githack.com/ajaxorg/ace/d52d9b32e5eb23a2a1981bf2b10344ede05b2214/kitchen-sink.html, if i press enter at the start of the file, the bug is indeed fixed, but if i type `*` at the start and only then press enter the comment indentation is applied.
We either need to test for a more specific pattern, or compute the state in the middle of line like outdent does.

Reply:
@marinsokol5 I could reproduce the issue highlighted by @nightwing, could you please take a look into it and update your PR to address it?